1.2.0 Release

nCastle1 opened the pull request.

nCastle1 committed:
WIP: completely remove blade view & UWP toolkit

nCastle1 committed:
Finishes fixing UI problems for 1903 target

nCastle1 committed:
Update title bar to match other styled panes

nCastle1 committed:
Add bookmarks and list-based TOC to UWP

nCastle1 committed:
Add TreeView-based TOC

nCastle1 committed:
Add TreeView-based TOC

nCastle1 committed:
Hide bookmarks when none are available

nCastle1 committed:
Merge branch 'ncastle/toc' of https://github.com/Esri/data-collection-dotnet into ncastle/toc

# Conflicts:
#	src/DataCollection.UWP/MainPage.xaml

nCastle1 committed:
Update TOC and bookmarks icons

nCastle1 committed:
Add TOC and bookmarks to WPF

nCastle1 committed:
Changes from review

nCastle1 committed:
Merge branch 'dev' into ncastle/toc

# Conflicts:
#	src/DataCollection.WPF/DataCollection.WPF (.NET Core).csproj

nCastle1 committed:
Fix issue with canceling taking a map offline

nCastle1 committed:
Doc, release notes, & image updates

nCastle1 committed:
Fix issue with WPF TOC

nCastle1 committed:
WIP: add legend, and move to Calcite icons

nCastle1 committed:
Clean up UWP icons

nCastle1 committed:
Fix UWP bug preventing sync cancel

nCastle1 committed:
Clean up icons WPF

nCastle1 committed:
Update release notes

nCastle1 committed:
Doc updates

nCastle1 committed:
Update doc images

nCastle1 committed:
Merge pull request #92 from Esri/ncastle/toc

Add TOC and bookmarks to data collection

nCastle1 committed:
Only show attachment icons on attachment tab

Note: also fixes an issue with the wrong icons being used in the origin related record popup

nCastle1 committed:
Improve appearance of accessory panels

nCastle1 committed:
Improve appearance of fields in WPF

nCastle1 committed:
Update the WPF side panel design to match UWP

nCastle1 committed:
Update the WPF side panel design to match UWP

nCastle1 committed:
Merge branch 'ncastle/ui-enhancements' of https://github.com/Esri/data-collection-dotnet into ncastle/ui-enhancements

# Conflicts:
#	src/DataCollection.WPF/ResourceDictionary.xaml

nCastle1 committed:
Adjust appearance of panel headers and tabs

nCastle1 committed:
Enhance WPF panel & overlay UI

nCastle1 committed:
Fix localization and layout issues WPF

nCastle1 committed:
Doc updates for 1.2 release

esreli commented on RELEASE.md:
```diff
@@ -1,5 +1,15 @@
 # Release notes
 
+## 1.2.0
```
I think we should get in the habit of putting periods at the end of each phrase or sentence in the RELEASE doc.

esreli replied:
I don't see a line that outlines the Legend. Can this be added?

esreli commented on docs/index.md:
```diff
@@ -44,9 +44,12 @@ The navigation bar's title reflects the name of the web map and the navigation b
-| ![Zoom To Location](/docs/images/zoom-to-location.png) | Zoom to user's location. |
-| ![Add Feature](/docs/images/add-feature.png) | Add a new spatial feature to map. |
+| ![Bookmarks icon](/src/DataCollection.UWP/Assets/Calcite/bookmark-32.png) | Bookmarks button reveals list of bookmarks in the map |
+| ![TOC icon](/src/DataCollection.UWP/Assets/Calcite/layers-32.png) | Contents button shows a list of layers, which can be toggled on and off |
```
Does the app make it clear this button is the "contents" button? If it's a direct TOC, perhaps we should just call it that?

esreli commented on docs/index.md:
```diff
@@ -44,9 +44,12 @@ The navigation bar's title reflects the name of the web map and the navigation b
-| ![Add Feature](/docs/images/add-feature.png) | Add a new spatial feature to map. |
+| ![Bookmarks icon](/src/DataCollection.UWP/Assets/Calcite/bookmark-32.png) | Bookmarks button reveals list of bookmarks in the map |
+| ![TOC icon](/src/DataCollection.UWP/Assets/Calcite/layers-32.png) | Contents button shows a list of layers, which can be toggled on and off |
+| ![Legend icon](/src/DataCollection.UWP/Assets/Calcite/legend-32.png) | Legend button shows the legend for the map |
```
```suggestion
| ![Legend icon](/src/DataCollection.UWP/Assets/Calcite/legend-32.png) | Legend button shows the layer symbols for the map |
```

esreli reviewed (changes requested):
It seems like we're in really good shape. I've reviewed this PR with a particular eye on documentation. Most obvious is doc found in:

- `docs/index.md`
- `README`
- `RELEASE`

Less obvious to me is the quality of in-line code commenting. Do you feel that modules are commented and difficult concepts are explained, in-line? I have to trust your judgement here.

nCastle1 committed:
Merge pull request #96 from Esri/ncastle/1-2-doc

Doc updates for 1.2 release

nCastle1 committed:
Merge pull request #95 from Esri/ncastle/ui-enhancements

UI enhancements

nCastle1 committed:
Update docs/index.md

Co-Authored-By: Eli Gregory <33433113+esreli@users.noreply.github.com>

nCastle1 committed:
Changes from 1.2.0 release review

nCastle1 committed:
Updated doc

esreli commented on docs/index.md:
```diff
@@ -91,6 +91,32 @@ If a user elects to delete the offline map, the app deletes the offline mobile m
+
+Web maps can include a list of bookmarks. Each bookmark consists of a map extent (visible area) and a name. Bookmarks can be authored in ArcGIS Pro and the ArcGIS Web Map Viewer.
+
+You can click the Bookmark icon to see a list of bookmarks in the map. Selecting a bookmark will navigate to the bookmark's extent.
```
Because "navigate" is an existing verb in the ArcGIS ecosystem, can we go with something else? Perhaps "set the map view's visible area to the bookmark's extent"?

esreli commented on docs/index.md:
```diff
@@ -91,6 +91,32 @@ If a user elects to delete the offline map, the app deletes the offline mobile m
 
 > A user can resume work online without deleting the offline map.
 
+### View and navigate to map bookmarks
```
Because "navigate" is an existing verb in the ArcGIS ecosystem, can we go with something else? Perhaps "View and set visible area from map bookmarks"? This probably isn't the best either, but perhaps it's on the right path..?

esreli reviewed (changes requested):
This new content is a great addition. I'm submitting just a few small notes. Thanks!

nCastle1 committed:
Update index.md

nCastle1 committed:
Misc. certification fixes

esreli reviewed (commented):
I think this is looking good. It's proven difficult for me to appropriately review this PR and as such my approval is conditional on a few items that I am unable to fully vet.

- [x] README.md (copy and images) is updated to reflect this version release.
- [x] RELEASE.md is updated to reflect this version release.
- [x] docs/index.md is updated to reflect this version release.
- [x] App's project version reflects this version release.
- [x] Discretion is applied for code that is better served by in-line code comments.
- [x] Rigorous ad-hoc testing guided by a test design document is performed and tests pass for all introduced features.
- [x] Loose ad-hoc testing of all general workflows guided by a test design document is performed and tests pass.

Of the above items one in particular stands out to me.

Can you elaborate on what measures have been taken to test features introduced in this version? From our internal planning repository, it appears the team has published a testing protocol and some functional tests for Bookmarks but not for TOC or Legend (or Layer Contents), as they related to Data Collection. 

I hope to ensure published releases are certified at minimum by a test design document agreed-upon by the team.

nCastle1 committed:
Update manifest and assembly versions

nCastle1 commented:
@esreli Re: inline comments, I've reviewed the code changes and I don't think anything needs additional inline comments; everything is either following an existing pattern or straightforward enough to read that inline comments aren't needed.

We don't have a test plan for most of what's new, so I'll work on that next.

esreli reviewed (approved):
https://github.com/Esri/data-collection-dotnet/pull/94#pullrequestreview-385853030

I submit my approval on the condition that all boxes are checked.  Specifically, because the Layer Contents test design document is merged into master, I believe the Data Collection 1.2 release for .NET can be certified at minimum by the supplied test design documents.

nCastle1 commented:
I've just tested one last time to confirm everything works. In addition to the rigoruous testing defined for the new layer content features, I've tested the following workflows in WPF (Framework), WPF (Core), and UWP:

* take map offline
    * Edit feature, sync, go back online and verify change synced
    * Add attachment, verify it is uploaded upon sync
    * Delete attachment, verify it is removed after sync
    * Add new feature, verify it is available online after sync
    * Attempt to add invalid attribute while offline, verified that that is prevented
* While online
    * Attempt to go offline when not logged in (auth dialog appears)
    * Attempt to go online when logged in
    * Add new feature; add new related record
    * Delete feature
    * Add attachment, open added attachment, delete attachment
    * Edit existing feature
    * Attempt to make invalid change (text in decimal field), verified saving is prevented
* Verified added, removed features and added, removed attachments are reflected when re-opening app (verified changes actually made it to feature service)
* Verified symbology updates when new inspections are added

Combined with the testing that was done for each constituent PR, I think this release has been sufficiently tested, so I'm going to go ahead and merge+release.
